src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/service/UserService.kt

```diff
@@ -96,6 +96,12 @@ class UserService(
     userPersonDetailRepository.findAll(UserSpecification(filter), pageRequest.withSort(::mapUserSummarySortProperties))
       .map { it.toUserSummary() }
 
+  @Transactional(readOnly = true)
+  fun findAllEmailAddressesByUsername(username: String): Set<String> =
+    userPersonDetailRepository.findById(username)
+      .map { user -> user.staff.emails.map { staffEmail -> staffEmail.email.lowercase() }.toSet() }
```

Comment:
Replicated the query found in [hmpps-auth.VerifyEmailService](https://github.com/ministryofjustice/hmpps-auth/blob/76db1eb76f7806a227393a334eecc7d00cb4eb66/src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/verify/VerifyEmailService.kt#L281-L288) which does a inner join between `internet_address where internet_address_class = 'EMAIL'` and `STAFF_USER_ACCOUNTS`

In nomis user roles we have entity mapping for `INTERNET_ADDRESSES` table to `EmailAddress` and `STAFF_USER_ACCOUNTS` to `UserPersonDetail`

We retrieve `UserPersonDetail` for that username which pulls in `Staff` and which has the following logic
```
@Where(clause = "OWNER_CLASS = 'STF' AND INTERNET_ADDRESS_CLASS = 'EMAIL'")
val emails: MutableList<EmailAddress> = mutableListOf()
```
So should pull in all of the email addresses as the query does